### ISSUE TITLE
[Bug] Android build fails

### ISSUE THREAD

**Issue by ashutoshbsathe** [mlc-ai/mlc-llm#1605]
## 🐛 Bug

Experimenting with [TinyLlama-1.1B-Chat-v1.0](https://huggingface.co/TinyLlama/TinyLlama-1.1B-Chat-v1.0) on Android. I was able to quantize and compile the model but `prepare_libs.sh` fails with following errors:

`clang++: error: unsupported argument 'native' to option '-march='` or `warning: onig_sys@69.8.1: clang-17: error: unsupported argument 'nocona' to option '-march='`. Unlike #1543, my build seems to be using `clang++` properly instead of `g++`

Detailed log: https://pastebin.com/1ZNZTure

More info about the compiler:

```
$TVM_NDK_CC --version
Android (10552028, +pgo, +bolt, +lto, -mlgo, based on r487747d) clang version 17.0.2 (https://android.googlesource.com/toolchain/llvm-project d9f89f4d16663d5012e5c09495f3b30ece3d2362)
Target: aarch64-unknown-linux-android34
Thread model: posix
InstalledDir: /home/ashutosh/HDD/Android/SDK/ndk/26.1.10909125//toolchains/llvm/prebuilt/linux-x86_64/bin

$TVM_NDK_CC -print-targets
  Registered Targets:
    aarch64    - AArch64 (little endian)
    aarch64_32 - AArch64 (little endian ILP32)
    aarch64_be - AArch64 (big endian)
    arm        - ARM
    arm64      - ARM64 (little endian)
    arm64_32   - ARM64 (little endian ILP32)
    armeb      - ARM (big endian)
    bpf        - BPF (host endian)
    bpfeb      - BPF (big endian)
    bpfel      - BPF (little endian)
    riscv32    - 32-bit RISC-V
    riscv64    - 64-bit RISC-V
    thumb      - Thumb
    thumbeb    - Thumb (big endian)
    wasm32     - WebAssembly 32-bit
    wasm64     - WebAssembly 64-bit
    x86        - 32-bit X86: Pentium-Pro and above
    x86-64     - 64-bit X86: EM64T and AMD64
```

## To Reproduce

1. Follow Android App guide from https://llm.mlc.ai/docs/deploy/android.html to compile and quantize the TinyLlama model
2. Run `./prepared_libs.sh` once everything is compiled

## Expected behavior

The runtime library and TVM bindings should be generated correctly

## Environment

 - Platform (e.g. WebGPU/Vulkan/IOS/Android/CUDA): Android
 - Operating system (e.g. Ubuntu/Windows/MacOS/...): Linux (6.6.10-arch1-1)
 - Device (e.g. iPhone 12 Pro, PC+RTX 3090, ...): Compiling on PC (RTX 3060) with target Android 14 (Nothing Phone 2)
 - How you installed MLC-LLM (`conda`, source): `pip`
 - How you installed TVM-Unity (`pip`, source): `pip`
 - Python version (e.g. 3.10): 3.10.0
 - GPU driver version (if applicable): 545.29.06
 - CUDA/cuDNN version (if applicable): 12.3
 - TVM Unity Hash Tag (`python -c "import tvm; print('\n'.join(f'{k}: {v}' for k, v in tvm.support.libinfo().items()))"`, applicable if you compile models): https://pastebin.com/jNU5EhRj
 - Any other relevant information: --

## Additional context

--

**Comment by ashutoshbsathe** [comment]
I tried manually setting the architecture in CMake and it did not work

**Comment by junrushao** [comment]
The `-march` issue is fixed in https://github.com/mlc-ai/mlc-llm/pull/1609

**Comment by ashutoshbsathe** [comment]
Thanks for the response.

Did a fresh `git clone --recursive` just to be sure I updated everything but the **issue still persists**. 
I even removed by TinyLlama config now and attempted compiling with default Llama and RedPajama models.

Additionally, I also get the following error now:
```
error: no member named '_V2' in namespace 'std::chrono'
using MLCTimePoint = std::chrono::_V2::system_clock::time_point;
                     ~~~~~~~~~~~~~^
```

**Comment by Kartik14** [comment]
Hey, This should be fixed by #1639
